storage/client.go

```diff
@@ -179,6 +179,44 @@ func (e UnexpectedStatusCodeError) Got() int {
 	return e.got
 }
 
+// NewClientFromConnectionString creates a Client from the connection string.
+func NewClientFromConnectionString(input string) (Client, error) {
+	var (
+		accountName, accountKey, endpointSuffix string
+		useHTTPS                                = defaultUseHTTPS
+	)
+
+	for _, pair := range strings.Split(input, ";") {
+		if pair == "" {
+			continue
+		}
+
+		equalDex := strings.IndexByte(pair, '=')
+		if equalDex <= 0 {
+			return Client{}, fmt.Errorf("Invalid connection segment %q", pair)
+		}
+
+		value := pair[equalDex+1:]
+		switch strings.ToLower(pair[:equalDex]) {
+		case "accountname":
+			accountName = value
+		case "accountkey":
+			accountKey = value
+		case "endpointsuffix":
+			endpointSuffix = value
+		case "defaultendpointsprotocol":
+			useHTTPS = value == "https"
+		default:
+			// ignore
```

Comment:
Is No-oping the correct decision here? I wound have thought it would be more appropriate to error out if an unrecognized key is encountered.

Reply:
@jhendrixMSFT / @mcardosos, mind taking a look to see if there are other possible values here?

Reply:
Looks like there are a bunch of other possible values
https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string

Reply:
Thanks for looking into it! I'll undo my change which errors in this case.